### PR TITLE
ci(backend): setup test workflow

### LINKED ISSUE #27: 🚄 CI: GitHub Action to run backend tests

### PR DESCRIPTION
**Pull request by CeeJayyy007**
### Description
- Setup GitHub action to run backend tests
 
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)\
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### Issue ticket number and link

Depends on: #39
Closes: #27 

### Checklist

- [x] Self-reviewed my code prior to submission.
- [ ] Tests have been added or updated (if applicable).
- [ ] Documentation has been updated (if applicable).
- [ ] Screenshots added (if applicable).
- [x] This PR is ready for review.

### CODE REVIEW

**Review by xamdoo — DISMISSED** [review]

File: .github/workflows/backend-tests.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    paths: ["backend/"]
+  pull_request:
+    branches: ["dev"]
+
+defaults:
+  run:
+    working-directory: "backend/"
+
+jobs:
+  run-frontend-tests:
```
Comment (RickyC0626) [inline .github/workflows/backend-tests.yml]: **suggestion:** `run-backend-tests`

**Review by RickyC0626 — CHANGES_REQUESTED** [review]
Could we also add `--coverage` flag in `test` script too?

File: .github/workflows/backend-tests.yml (hunk 1 above)
Comment (CeeJayyy007) [reply .github/workflows/backend-tests.yml]: My bad. Edit oversight. Will update accordingly

Comment (CeeJayyy007) [reply .github/workflows/backend-tests.yml]: Update, please review

**Review by RickyC0626 — APPROVED** [review]

**Review by xamdoo — APPROVED** [review]

**Pull request merged**